frontend/spec/features/products_spec.rb

```diff
@@ -20,6 +20,21 @@
     expect(page).to have_content("Shopping Cart")
   end
 
+  # Regression spec for Spree [PR#7442](https://github.com/spree/spree/pull/7442)
+  context "when generating product links" do
+    let(:product) { Spree::Product.available.first }
+
+    it "should not use the *_url helper to generate the product links" do
+      visit spree.root_path
+      expect(page).not_to have_xpath(".//a[@href='#{spree.product_url(product, host: default_url_options[:host])}']")
```

Comment:
Turns out "current_host" is a thing, so yeah definitely simpler this way. Thanks!